src/sentry/static/sentry/app/views/settings/organizationAuth/organizationAuthList.jsx

```diff
@@ -14,23 +14,45 @@ import getCookie from 'app/utils/getCookie';
 
 import ProviderItem from './providerItem';
 
-class OrganizationAuthList extends React.Component {
-  static contextTypes = {
-    organization: SentryTypes.Organization,
-  };
+const providerPopularity = {
+  google: 0,
+  github: 1,
+  okta: 2,
+  'active-directory': 3,
+  saml2: 4,
+  onelogin: 5,
+  rippling: 6,
+  auth0: 7,
+};
 
+class OrganizationAuthList extends React.Component {
   static propTypes = {
     providerList: PropTypes.arrayOf(SentryTypes.AuthProvider).isRequired,
     activeProvider: PropTypes.object,
   };
 
+  static contextTypes = {
+    organization: SentryTypes.Organization,
+  };
+
   render() {
     const {activeProvider} = this.props;
     const {organization} = this.context;
     const features = organization.features;
 
-    // Sort feature-flagged integrations last
-    const providerList = (this.props.providerList || []).sort((a, b) => {
+    // Sort provider list twice: first, by popularity,
+    // and then a second time, to sort unavailable providers for the current plan to the end of the list.
+    const sortedByPopularity = (this.props.providerList || []).sort((a, b) => {
+      if (!(a.key in providerPopularity)) {
+        return -1;
+      }
+      if (!(b.key in providerPopularity)) {
+        return 1;
+      }
+      return providerPopularity[a.key] > providerPopularity[b.key];
```

Comment:
should return explicit numbers instead of a boolean. For example:

```suggestion
      return providerPopularity[a.key] > providerPopularity[b.key] ? 1 : -1;
```